socorro/cron/app.py

```diff
@@ -0,0 +1,1207 @@
+                    'index': namespace_index
+                }
+                namespace_name = template_for_namespace % namespace_name_dict
+                subordinate_namespace_names.append(namespace_name)
+                # create the new Namespace
+                required_config.namespace(namespace_name)
+                a_class_namespace = required_config[namespace_name]
+                # add options for the 'extra data'
+                try:
+                    extra_options = extra_extractor(class_list_element)
+                    a_class_namespace.update(extra_options)
+                except NotImplementedError:
+                    pass
+                # add options frr the classes required config
+                try:
+                    for k, v in a_class.get_required_config().iteritems():
+                        if k not in reference_namespace:
+                            a_class_namespace[k] = v
+                except AttributeError:  # a_class has no get_required_config
+                    pass
+
+            @classmethod
+            def to_str(cls):
+                """this method takes this inner class object and turns it back
+                into the original string of classnames.  This is used
+                primarily as for the output of the 'help' option"""
+                return cls.original_input
+
+        return InnerClassList  # result of class_list_converter
+    return class_list_converter  # result of classes_in_namespaces_converter
+
+
+def get_extra_as_options(input_str):
+    if '|' not in input_str:
+        raise JobDescriptionError('No frequency and/or time defined')
+    metadata = input_str.split('|')[1:]
+    if len(metadata) == 1:
+        if ':' in metadata[0]:
+            frequency = '1d'
+            time_ = metadata[0]
+        else:
+            frequency = metadata[0]
+            time_ = None
+    else:
+        frequency, time_ = metadata
+
+    n = Namespace()
+    n.add_option(
+        'frequency',
+        doc='frequency',
+        default=frequency,
+        exclude_from_print_conf=True,
+        exclude_from_dump_conf=True
+    )
+    n.add_option(
+        'time',
+        doc='time',
+        default=time_,
+        exclude_from_print_conf=True,
+        exclude_from_dump_conf=True
+    )
+    return n
+
+
+def check_time(value):
+    """check that it's a value like 03:45 or 1:1"""
+    try:
+        h, m = value.split(':')
+        h = int(h)
+        m = int(m)
+        if h >= 24 or h < 0:
+            raise ValueError
+        if m >= 60 or m < 0:
+            raise ValueError
+    except ValueError:
+        raise TimeDefinitionError("Invalid definition of time %r" % value)
+
+
+def line_splitter(text):
+    return [x.strip() for x in re.split('\n|,|;', text.strip())
+            if x.strip() and not x.strip().startswith('#')]
+
+
+def pipe_splitter(text):
+    return text.split('|', 1)[0]
+
+
+class CronTabberBase(RequiredConfig):
```

Comment:
We can (and should) collapse this with `CronTabberApp`. We don't need to have both. Also, we can probably take out commands we don't use like "version"